container_registry/container_analysis/samples_test.py

```diff
@@ -0,0 +1,178 @@
+#!/bin/python
+# Copyright 2018 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from os import environ
+from time import sleep, time
+
+from google.api_core.exceptions import InvalidArgument
+from google.api_core.exceptions import NotFound
+from google.cloud.devtools.containeranalysis_v1alpha1.proto.\
+    containeranalysis_pb2 import Occurrence
+from google.cloud.devtools.containeranalysis_v1alpha1.proto.\
+    package_vulnerability_pb2 import VulnerabilityType
+from google.cloud.pubsub import SubscriberClient
+
+import samples
+
+PROJECT_ID = environ['GCLOUD_PROJECT']
+SLEEP_TIME = 1
+TRY_LIMIT = 20
+
+
+class TestContainerAnalysisSamples:
```

Comment:
Prefer just having test methods, [see this example](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/vision/cloud-client/detect/detect_test.py) for a pattern you can use.

Reply:
Does this pattern work well with setup/teardown functions? I'm having trouble finding examples

Reply:
[Here is an example](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/iot/api-client/http_example/cloudiot_http_example_test.py#L45-L54) where the fixture yields the dependency before the test and destroys it after.